bpfd/src/storage.rs

```diff
@@ -1,206 +1,370 @@
+        }));
+    }
+
+    async fn probe(
+        &self,
+        _request: Request<ProbeRequest>,
+    ) -> Result<Response<ProbeResponse>, Status> {
+        return Ok(Response::new(ProbeResponse {
+            ..Default::default()
+        }));
+    }
+
+    async fn get_plugin_capabilities(
+        &self,
+        _request: Request<GetPluginCapabilitiesRequest>,
+    ) -> Result<Response<GetPluginCapabilitiesResponse>, Status> {
+        return Ok(Response::new(GetPluginCapabilitiesResponse {
+            capabilities: vec![PluginCapability {
+                r#type: Some(plugin_capability::Type::Service(
+                    plugin_capability::Service {
+                        r#type: plugin_capability::service::Type::ControllerService.into(),
+                    },
+                )),
+            }],
+        }));
+    }
+}
 
 #[async_trait]
 impl Controller for CsiController {
     async fn create_volume(
         &self,
         _request: Request<CreateVolumeRequest>,
     ) -> std::result::Result<Response<CreateVolumeResponse>, Status> {
-        todo!()
+        return Err(Status::unimplemented(""));
     }
 
     async fn delete_volume(
         &self,
         _request: Request<DeleteVolumeRequest>,
     ) -> std::result::Result<Response<DeleteVolumeResponse>, Status> {
-        todo!()
+        return Err(Status::unimplemented(""));
     }
+
     async fn controller_publish_volume(
         &self,
         _request: Request<ControllerPublishVolumeRequest>,
     ) -> std::result::Result<Response<ControllerPublishVolumeResponse>, Status> {
-        todo!()
+        return Err(Status::unimplemented(""));
     }
+
     async fn controller_unpublish_volume(
         &self,
         _request: Request<ControllerUnpublishVolumeRequest>,
     ) -> std::result::Result<Response<ControllerUnpublishVolumeResponse>, Status> {
-        todo!()
+        return Err(Status::unimplemented(""));
     }
+
     async fn validate_volume_capabilities(
         &self,
         _request: Request<ValidateVolumeCapabilitiesRequest>,
     ) -> std::result::Result<Response<ValidateVolumeCapabilitiesResponse>, Status> {
-        todo!()
+        return Err(Status::unimplemented(""));
     }
+
     async fn list_volumes(
         &self,
         _request: Request<ListVolumesRequest>,
     ) -> std::result::Result<Response<ListVolumesResponse>, Status> {
-        todo!()
+        return Err(Status::unimplemented(""));
     }
+
     async fn get_capacity(
         &self,
         _request: Request<GetCapacityRequest>,
     ) -> std::result::Result<Response<GetCapacityResponse>, Status> {
-        todo!()
+        return Err(Status::unimplemented(""));
     }
+
     async fn controller_get_capabilities(
         &self,
         _request: Request<ControllerGetCapabilitiesRequest>,
     ) -> std::result::Result<Response<ControllerGetCapabilitiesResponse>, Status> {
-        todo!()
+        return Ok(Response::new(ControllerGetCapabilitiesResponse {
```

Comment:
Do we need to implement this at all?
Would be convenient if we didn't need to as we could drop the Controller impl entirely.